win32ss/user/ntuser/sysparams.c

```diff
@@ -1583,7 +1583,17 @@ SpiGetSet(UINT uiAction, UINT uiParam, PVOID pvParam, FLONG fl)
             return SpiSetUserPref(UPM_LISTBOXSMOOTHSCROLLING, pvParam, fl);
 
         case SPI_GETGRADIENTCAPTIONS:
-            return SpiGetUserPref(UPM_GRADIENTCAPTIONS, pvParam, fl);
+        {
+            if (NtGdiGetDeviceCaps(ScreenDeviceContext, BITSPIXEL) <= 8)
```

Comment:
Calling NtGdi functions while holding the user lock is a slippery slope. But in this particular case it seems that this is safe.